url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/util/Queue.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Java Settlers - An online multiplayer version of the game Settlers of Catan
+ * Copyright (C) 2003  Robert S. Thomas
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ *
+ * The author of this program can be reached at thomas@infolab.northwestern.edu
+ **/
+package soc.util;
+
+import java.util.Vector;
+
+
+/**
+ * DOCUMENT ME!
+ *
+ * @author $author$
+ * @version $Revision: 1.1 $
+ */
+public class Queue
+{
+    // Internal storage for the queue'd objects
+    private Vector vec = new Vector();
+
+    /**
+     * DOCUMENT ME!
+     *
+     * @param o DOCUMENT ME!
+     */
+    synchronized public void put(Object o)
+    {
+        //D.ebugPrintln(">put-> "+o);
+        // Add the element
+        vec.addElement(o);
+
+        // There might be threads waiting for the new object --
+        // give them a chance to get it
+        notifyAll();
+    }
+
+    /**
+     * DOCUMENT ME!
+     *
+     * @return DOCUMENT ME!
+     */
+    synchronized public Object get()
+    {
+        while (true)
+        {
+            if (vec.size() > 0)
+            {
+                // There's an available object!
+                Object o = vec.elementAt(0);
+
+                //D.ebugPrintln("<-get< "+o); 
+                // Remove it from our internal list, so someone else
+                // doesn't get it.
+                vec.removeElementAt(0);
+
+                // Return the object
+                return o;
+            }
+            else
+            {
+                // There aren't any objects available.  Do a wait(),
+                // and when we wake up, check again to see if there
+                // are any.
+                try
+                {
+                    wait();
+                }
+                catch (InterruptedException ie) {}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)